openrtb_ext/request_wrapper.go

```diff
@@ -1265,6 +1439,17 @@ func (w *ImpWrapper) rebuildImpExt() error {
 	return nil
 }
 
+func (w *ImpWrapper) Clone() *ImpWrapper {
+	if w == nil {
+		return nil
```

Comment:
I see that `ImpWrapper` has an embedded type `*openrtb2.Imp`. I am curious how does that work in this case. Do we not need to explicitly copy that object over too?

Reply:
IMHO this code is cloning the state of the RequestWrapper. The embedded imp pointer is part of the Request itself.  I believe it's good to keep with normal usage semantics, where the caller code is responsible for copying the Request before making a change. In most cases, the RequestWrapper assists only with the ext fields and doesn't overrite shared memory.